### PR TITLE
prov/rxm: Clean up recv_list/unexp_msg & code style

### PR DESCRIPTION
**Pull request by nikitaxgusev**

### CODE REVIEW

**Comment by nikitaxgusev** [comment]
See segfault on fi_recv_cancel test in rxd_cleanup_unexp_msg (in dlist_remove).WIP on invsetigating.

**Comment by nikitaxgusev** [comment]
The issue is that we are cleaning up recv_list before rxm_recv_queue_close. 
But we are NOT cleaning unexp_msg_list.

File: prov/rxm/src/rxm_ep.c [hunk 1]
```diff
@@ -327,7 +327,13 @@ static void rxm_recv_queue_close(struct rxm_recv_queue *recv_queue)
...
 		rxm_recv_fs_free(recv_queue->fs);
 	}
-	// TODO cleanup recv_list and unexp msg list
+
+	while (!dlist_empty(&recv_queue->recv_list)) {
+		dlist_remove(&recv_queue->recv_list);
+	}
+	while (!dlist_empty(&recv_queue->unexp_msg_list)) {
+		dlist_remove(&recv_queue->unexp_msg_list);
+	}
```
Comment (aingerson) [inline prov/rxm/src/rxm_ep.c]: Try moving this code to before the rxm_recv_fs_free call. The entries that are stored in the dlist are part of the freestack that is freed there

**Comment by jswaro** [comment]
I am seeing a massive amount of segmentation faults with this change. Are you seeing them in your testing?

**Comment by nikitaxgusev** [comment]
@jswaro yes, I am working on resolving the issue. Thanks.
Sorry, for this miss-testing before making this PR.

**Comment by nikitaxgusev** [comment]
@jswaro @shefty Locally fi_av_test doesn't fail with this change. I am not sure, if this change related to the fail.

**Comment by jswaro** [comment]
```
[2019-11-20T09:44:40.238Z] - name:   fi_av_test -g 10.0.0.2 -n 1 -s 10.0.0.2 -p "verbs;ofi_rxm"

[2019-11-20T09:44:40.238Z]   result: Fail

[2019-11-20T09:44:40.238Z]   time:   0

[2019-11-20T09:44:40.238Z]   server_cmd: /scratch/jenkins/jenkins-root/workspace/ibfabric_OFIWG_Libfabric_PR-5449/installs/fabtests/bin/fi_av_test -g 10.0.0.2 -n 1 -s 10.0.0.2 -p "verbs;ofi_rxm"

[2019-11-20T09:44:40.238Z]   server_stdout: |

[2019-11-20T09:44:40.238Z]     Testing AVs on fabric IB-0xfe80000000000000

[2019-11-20T09:44:40.238Z]     

[2019-11-20T09:44:40.238Z]     Testing with type = FI_AV_MAP

[2019-11-20T09:44:40.238Z]     Running av_open_close [Test open and close AVs of varying sizes]...PASS!

[2019-11-20T09:44:40.238Z]     Running av_good_sync [Test sync AV insert with good address]...PASS!

[2019-11-20T09:44:40.238Z]     Running av_null_fi_addr [Test AV insert without specifying fi_addr]...skipped because: test not valid for AV type FI_AV_MAP

[2019-11-20T09:44:40.238Z]     Running av_good_vector_async [Test async AV insert with vector of good addresses]...PASS!

[2019-11-20T09:44:40.238Z]     Running av_zero_async [Test async insert AV insert of zero addresses]...PASS!

[2019-11-20T09:44:40.238Z]     Running av_good_2vector_async [Test async AV inserts with two address vectors]...PASS!

[2019-11-20T09:44:40.585Z]     Running av_insert_stages [Test AV insert at various stages]...
```
Looks like there wasn't any output for av_insert_stages, or it couldn't determine the ret code. That might be the place to start.

**Comment by nikitaxgusev** [comment]
@jswaro I get it, but I can't reproduce it by hands.

**Pull request closed without merging**